app/models/physical_chassis.rb

```diff
@@ -7,10 +7,18 @@ class PhysicalChassis < ApplicationRecord
   belongs_to :ext_management_system, :foreign_key => :ems_id, :inverse_of => :physical_chassis,
     :class_name => "ManageIQ::Providers::PhysicalInfraManager"
   belongs_to :physical_rack, :foreign_key => :physical_rack_id, :inverse_of => :physical_chassis
+  belongs_to :parent_physical_chassis,
+             :class_name => "PhysicalChassis",
+             :inverse_of => :child_physical_chassis
 
   has_many :event_streams, :inverse_of => :physical_chassis, :dependent => :nullify
   has_many :physical_servers, :dependent => :destroy, :inverse_of => :physical_chassis
   has_many :physical_storages, :dependent => :destroy, :inverse_of => :physical_chassis
+  has_many :child_physical_chassis,
+           :class_name  => "PhysicalChassis",
+           :dependent   => :destroy,
```

Comment:
@tadeboro this should probably be `:dependent => :nullify` since the child chassis will have their own ems_id+ems_ref and thus should be deleted by ems_refresh.
E.g. what if the child chassis were removed from the parent chassis and the parent was deleted but the children were still around.

Reply:
I never thought about this, since in Redfish world, removing intermediary chassis would be so disruptive, that most (if not all) `ems_ref`s would change. But you are correct, reparenting/deleting should be handled  by refresh process. Will fix this.